src/librustc_borrowck/borrowck/gather_loans/mod.rs

```diff
@@ -289,6 +291,30 @@ impl<'a, 'tcx> GatherLoanCtxt<'a, 'tcx> {
         }
     }
 
+    fn loan_scope_from_region(&self, loan_region: ty::Region<'tcx>) -> Option<region::CodeExtent> {
```

Comment:
Does this make sense? `Given a region for a loan, returns the corresponding loan_scope that could represent the gen_scope and/or kill_scope of a loan.` It's the exact same code that used to be in lines 351-374 in the base file. Based on the usage, it made sense to me that this would represent the combination of the gen_scope and kill_scope for what I called a SafeLoan (do you think there's a better name?). For normal loans, it's a possible value for the gen_scope and/or kill_scope. (See `compute_gen_scope` and `compute_kill_scope`).